src/pages/views/PageDetails.tsx

```diff
@@ -59,22 +74,99 @@ export const PageDetails: React.FC<PageDetailsProps> = ({ id, params }) => {
     }
   };
 
+  const [
+    deleteAttributeValue,
+    deleteAttributeValueOpts
+  ] = useAttributeValueDeleteMutation({});
+
   return (
     <TypedPageRemove variables={{ id }} onCompleted={handlePageRemove}>
       {(pageRemove, pageRemoveOpts) => (
         <TypedPageUpdate>
           {(pageUpdate, pageUpdateOpts) => (
             <TypedPageDetailsQuery variables={{ id }}>
               {pageDetails => {
-                const handleUpdate = async (data: PageData) => {
+                const handleUpdate = async (data: PageSubmitData) => {
+                  let errors: Array<
+                    | AttributeErrorFragment
+                    | UploadErrorFragment
+                    | PageErrorFragment
+                  > = [];
+
+                  const uploadFilesResult = await Promise.all(
+                    data.attributesWithNewFileValue.map(fileAttribute =>
+                      uploadFile({
+                        variables: {
+                          file: fileAttribute.value
+                        }
+                      })
+                    )
+                  );
+
+                  const attributesWithAddedNewFiles: AttributeValueInput[] = uploadFilesResult.reduce(
```

Comment:
Same as above - I think this could be map? Also would move to a separate function, that could return both `attributesWithAddedNewFiles` and connected errors